tests/artifact/test_bento_service_artifact.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+from bentoml.artifact import BentoServiceArtifact
+
+
+def test_valid_artifact_name():
+    name = "_test"
+    artifact = BentoServiceArtifact(name)
+    assert artifact.name == "_test"
+
+    name = "test"
+    artifact = BentoServiceArtifact(name)
+    assert artifact.name == "test"
+
+    name = "TEST00"
+    artifact = BentoServiceArtifact(name)
+    assert artifact.name == "TEST00"
+
+    name = "__"
+    artifact = BentoServiceArtifact(name)
+    assert artifact.name == "__"
+
+    name = "thisIs012Along__erTest"
+    artifact = BentoServiceArtifact(name)
+    assert artifact.name == "thisIs012Along__erTest"
+
+
+def test_unvalid_artifact_name():
+    name = ""
+    with pytest.raises(ValueError) as e:
+        BentoServiceArtifact(name)
+    assert "'name' must start with a letter" in str(e.value)
+
+    name = "_"
```

Comment:
Looks like this case does fit the requirement description: `'name' must start with a letter (a-z|A-Z) or underscore character and can only contain alpha-numeric characters and underscores`

Could you take a look and see if the `benotoml.utils.isidentifier` method work for this check?

Reply:
Actually, since we dropped support for PY2, we can probably just use the `name.isidentifier()` here. The error message could be improved as well, e.g.:

```
Artifact name must be a valid python identifier, a string is considered a valid identifier if it only contains alphanumeric letters (a-z) and (0-9), or underscores (_). A valid identifier cannot start with a number, or contain any spaces.
```

Reply:
I changed the regex to `name.isidentifier()` and the error message with the one you suggested. 
Btw, `benotoml.utils.isidentifier` function was failing for the check and the check with accents (same for `name.isidentifier()`).
Those, 2 tests were not fitting the requirements of a python identifier. I was a bit to restrictive when I read the definition and actually you can define variables names with accents or using the Underscore(_) character even if it is not recommended.

Reply:
Thanks for updating the PR @AlexDut, surprised that characters with accents work for `isidentifier`, but I think technology as long as it is a valid python identifier we can use it as a property name for accessing the artifact.